hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ContainerKeyScanner.java

```diff
@@ -0,0 +1,372 @@
+      OmMetadataManagerImpl omMetadataManager) throws IOException {
+    long keysProcessed = 0L;
+
+    // Anything but not FSO bucket layout
+    Table<String, OmKeyInfo> fileTable = omMetadataManager.getKeyTable(
+        BucketLayout.DEFAULT);
+    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+             iterator = fileTable.iterator()) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> next = iterator.next();
+        keysProcessed++;
+
+        if (Objects.isNull(next.getValue().getKeyLocationVersions())) {
+          continue;
+        }
+
+        processKeyData(containerKeyInfos, next.getKey(), next.getValue());
+      }
+    }
+
+    return keysProcessed;
+  }
+
+
+  private long processFileTable(
+      Map<Long, List<ContainerKeyInfo>> containerKeyInfos,
+      OmMetadataManagerImpl omMetadataManager)
+      throws IOException {
+    long keysProcessed = 0L;
+
+    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+             iterator = omMetadataManager.getFileTable().iterator()) {
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> next = iterator.next();
+        keysProcessed++;
+
+        if (Objects.isNull(next.getValue().getKeyLocationVersions())) {
+          continue;
+        }
+
+        processFileData(containerKeyInfos, next.getKey(), next.getValue(),
+            omMetadataManager);
+      }
+    }
+
+    return keysProcessed;
+  }
+
+  /**
+   * @param key file table key.
+   * @return Pair of volume id and bucket id.
+   */
+  private Pair<Long, Long> parseKey(String key) {
+    String[] keyParts = key.split(OM_KEY_PREFIX);
+    return Pair.of(Long.parseLong(keyParts[1]), Long.parseLong(keyParts[2]));
+  }
+
+  private void processKeyData(
+      Map<Long, List<ContainerKeyInfo>> containerKeyInfos,
+      String key, OmKeyInfo keyInfo) {
+    long volumeId = 0L;
+    long bucketId = 0L;
+
+    for (OmKeyLocationInfoGroup locationInfoGroup :
+        keyInfo.getKeyLocationVersions()) {
+      for (List<OmKeyLocationInfo> locationInfos :
+          locationInfoGroup.getLocationVersionMap().values()) {
+        for (OmKeyLocationInfo locationInfo : locationInfos) {
+          if (containerIds.contains(locationInfo.getContainerID())) {
+
+            containerKeyInfos.merge(locationInfo.getContainerID(),
+                new ArrayList<>(Collections.singletonList(
+                    new ContainerKeyInfo(locationInfo.getContainerID(),
+                        keyInfo.getVolumeName(), volumeId,
+                        keyInfo.getBucketName(), bucketId, keyInfo.getKeyName(),
+                        keyInfo.getParentObjectID()))),
+                (existingList, newList) -> {
+                  existingList.addAll(newList);
+                  return existingList;
+                });
+          }
+        }
+      }
+    }
+  }
+
+  private void processFileData(
```

Comment:
> We recently increased chars per line to 120. Please update your IDE setting if you have not.

Thanks for this! Somehow I've missed this update.

> Use early return to avoid nesting.

Fixed.

> I think getDirectoryTableData can be loaded outside the loops and no need to use isDirTableLoaded to check if it is loaded or not.

You are right, however, I used that to enable quick lazy loading mechanism.

> If it is to load only when needed, may be use some other way to load lazily.

Do you have any suggestions for this?